src/workerd/io/actor-cache.c++

```diff
@@ -2581,7 +2582,10 @@ kj::Promise<void> ActorCache::flushImplUsingSinglePut(PutFlush putFlush) {
   // See the comment in flushImplUsingTxn for why we need to construct our RPC and then wait on
   // reads before actually sending the write. The same exact logic applies here.
   co_await waitForPastReads();
-  co_await request.send().ignoreResult();
+  {
+    durationAlert(clock, kj::Duration(kj::SECONDS*1), "storage operation took longer than 1s: single put");
```

Comment:
I'm not sure this does what you're expecting...  I suspect the compiler is allowed to instantiate a temporary "durationAlert" value for the scope of the statement, then free it before doing the `co_await`.  If you want the value to exist for the scope of the code block, I think some further action is needed, like putting it in a named variable?

I guess one way to test this would be to add a `usleep(5000 * 1000);` statement in the block and see if the log fires.  Or alternately, to put in several `KJ_DBG()` statements to verify that the destructor runs after the end of the code block.

Reply:
This indeed doesn't do what I want it to do. Tried to do something similar in a separate c++ experiment, and I do indeed need to bind it to a named variable. Thanks for the feedback!